libraries/AP_HAL_Linux/RCInput_RCProtocol.cpp

```diff
@@ -48,48 +48,29 @@ int RCInput_RCProtocol::open_sbus(const char *path)
         fd = -1;
         return -1;
     }
-    tio.c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR
-                     | IGNCR | ICRNL | IXON);
-    tio.c_iflag |= (INPCK | IGNPAR);
-    tio.c_oflag &= ~OPOST;
-    tio.c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
-    tio.c_cflag &= ~(CSIZE | CRTSCTS | PARODD | CBAUD);
-    // use BOTHER to specify speed directly in c_[io]speed member
-    tio.c_cflag |= (CS8 | CSTOPB | CLOCAL | PARENB | BOTHER | CREAD);
-    tio.c_ispeed = 100000;
-    tio.c_ospeed = 100000;
-    if (ioctl(fd, TCSETS2, &tio) != 0) {
-        close(fd);
-        fd = -1;
-        return -1;
-    }
+    if(is_sbus) {
+        // SBUS configuration
+        tio.c_iflag &= ~(IGNBRK | BRKINT | PARMRK | ISTRIP | INLCR
+                         | IGNCR | ICRNL | IXON);
+        tio.c_iflag |= (INPCK | IGNPAR);
+        tio.c_oflag &= ~OPOST;
+        tio.c_lflag &= ~(ECHO | ECHONL | ICANON | ISIG | IEXTEN);
+        tio.c_cflag &= ~(CSIZE | CRTSCTS | PARODD | CBAUD);
+        // use BOTHER to specify speed directly in c_[io]speed member
+        tio.c_cflag |= (CS8 | CSTOPB | CLOCAL | PARENB | BOTHER | CREAD);
+        tio.c_ispeed = 100000;
+        tio.c_ospeed = 100000;
+    } else {
+        // Other (SRXL, DSM, etc) configuration
+        tio.c_cflag &= ~(PARENB|CSTOPB|CSIZE|CBAUD);
+        tio.c_cflag |= CS8 | CREAD | CLOCAL | B115200;
 
-    return fd;
-}
 
-/*
-  open a 115200 UART
- */
-int RCInput_RCProtocol::open_115200(const char *path)
-{
-    int fd = open(path, O_RDWR | O_NONBLOCK | O_CLOEXEC);
-    if (fd == -1) {
-        return -1;
+        tio.c_lflag &= ~(ICANON|ECHO|ECHOE|ISIG);
+        tio.c_iflag &= ~(IXON|IXOFF|IXANY);
+        tio.c_iflag &= ~(INLCR|ICRNL|IGNCR|IUCLC|BRKINT);
+        tio.c_oflag &= ~OPOST;
```

Comment:
all flags should be the same for all protocols, except baudrate and stopbits. And even for baudrate you can use `c_ispeed = 115200` rather than the old B115200 in `c_cflag`

Reply:
I am not an expert on SBUS, and I don't have a way to test it. I think there are some changes to the SRXL setup that should go to SBUS, but I don't want to change it without testing it. There are more differences between the setups than just stopbit and baud rate.